fix: Enable real-time extra links updates for TriggerDagRunOperator

subhash-0000 opened the pull request:
Fixes #58928 

The "Triggered DAG" button from `TriggerDagRunOperator` now appears in real-time without requiring a manual page refresh.

### Problem
Previously, extra links (including the "Triggered DAG" button) only appeared after:
1. The parent task completed
2. User manually refreshed the browser page

This reduced the utility of extra links, especially for monitoring triggered child DAGs in real-time.

### Solution
- Add smart polling to `ExtraLinks` component using the existing `useAutoRefresh` hook
- Poll automatically when DAG is active and links are not yet available
- Stop polling once extra links appear to reduce server load
- Respects global `auto_refresh_interval` config and DAG paused state

### Changes Made
- Modified `airflow-core/src/airflow/ui/src/pages/TaskInstance/ExtraLinks.tsx`
- Added `refetchInterval` logic with conditional polling
- Follows the same pattern used in `Details.tsx` for polling task instances
- No breaking changes, no new dependencies

### Testing

#### Code Quality:
- [x] TypeScript types are correct
- [x] Build succeeds without errors (`vite build` passes)
- [x] Follows existing patterns (`useAutoRefresh` hook)
- [x] No new dependencies added
- [x] Clear inline comments explain the logic

#### Manual Testing Checklist (for reviewers):
- [ ] Start Breeze with `breeze start-airflow --dev-mode`
- [ ] Create DAG with `TriggerDagRunOperator`
- [ ] Trigger the parent DAG
- [ ] Verify "Triggered DAG" button appears immediately (no refresh needed)
- [ ] Verify button remains visible while child DAG runs
- [ ] Verify polling stops once button appears (check Network tab in DevTools)
- [ ] Verify respects DAG paused state

### Performance Impact
✅ **Positive**: Polling stops when links appear, actually reducing server load compared to continuous polling  
✅ Respects `auto_refresh_interval` config (user-configurable)  
✅ Only polls when DAG is active (not paused)

### Related
This fix uses the same auto-refresh pattern already established in `Details.tsx` for task instance polling, ensuring consistency and maintainability across the codebase.

---

**Type**: Bug Fix  
**Component**: UI (React)  
**Breaking Changes**: None  
**New Dependencies**: None

subhash-0000 committed:
fix: Enable real-time extra links updates for TriggerDagRunOperator

- Add smart polling to ExtraLinks component using useAutoRefresh hook
- Poll automatically when DAG is active and links are not yet available
- Stop polling once extra links appear to reduce server load
- Respects global auto_refresh_interval config and DAG paused state
- Fixes issue where Triggered DAG button only appeared after manual refresh

Closes #58928

bbovenzi commented on airflow-core/src/airflow/ui/src/pages/TaskInstance/ExtraLinks.tsx:
```diff
@@ -21,17 +21,33 @@ import { useTranslation } from "react-i18next";
+          query.state.data?.extra_links && Object.keys(query.state.data.extra_links).length > 0;
+
+        // Continue polling if auto-refresh is enabled and we don't have links yet
+        return !hasLinks && refetchInterval ? refetchInterval : false;
```
This will poll forever if the task doesn't have extra links. Lets also call `useTaskServiceGetTask` and see if `task.extraLinks.length > 0` first.

We probably want to make that change in `TaskInstance/Details.tsx` to check if we even want to render the extra links component at all

subhash-0000 committed:
fix: prevent infinite polling for tasks without extra links

- Add task metadata check using useTaskServiceGetTask
- Only poll if task.extra_links.length > 0
- Conditionally render ExtraLinks component in Details.tsx
- Prevents unnecessary API calls for tasks without extra links

Addresses review feedback from @bbovenzi

pierrejeambrun commented on airflow-core/src/airflow/ui/src/pages/TaskInstance/Details.tsx:
```diff
@@ -95,7 +95,7 @@ export const Details = () => {
         />
       )}
-      <ExtraLinks />
+      {taskInstance?.extra_links && taskInstance.extra_links.length > 0 ? <ExtraLinks /> : undefined}
```
This doesn't exist on a TaskInstanceReponse. (`extra_links`).

pierrejeambrun reviewed (commented):
An easy way is to re-use the same parent `refetchInterval: (query) => (isStatePending(query.state.data?.state) ? refetchInterval : false),`

That is based on the 'try' 'pending' status. You can give that to the child component so it will refresh as long as the try isn't settled. Which might save you one 'useTaskServiceGetTask` call as well.

subhash-0000 committed:
refactor: reuse parent refetchInterval pattern in child components

- Pass parent's refetchInterval to ExtraLinks and BlockingDeps components
- Remove duplicate useAutoRefresh and useTaskServiceGetTask calls
- Eliminate invalid taskInstance.extra_links property access (doesn't exist on TaskInstanceResponse)
- Polling now tied to try instance pending state - stops automatically when task completes
- Simplifies architecture by centralizing refresh logic in parent component

Addresses review feedback from @pierrejeambrun and @bbovenzi

subhash-0000 committed:
style: apply pre-commit hook formatting fixes

- Reorder type union to alphabetical order (number | false)
- Format ExtraLinks component to single line
